internal/apex/apex.go

```diff
@@ -331,13 +331,23 @@ func (ax *Apex) Reconcile(zoneID uuid.UUID, firstTime bool) error {
 			newPeers = append(newPeers, p)
 		}
 	}
+
 	if changed {
 		ax.logger.Debugf("Peers listing has changed, recalculating configuration")
 		ax.buildPeersConfig()
 		if err := ax.DeployWireguardConfig(newPeers, false); err != nil {
 			return err
 		}
 	}
+
+	if len(ax.peerCache) > len(peerListing) {
```

Comment:
I think it's possible to miss a delete here. IIRC, this is currently based on polling, right? A peer could have been deleted no matter what the lengths of these lists are compared to each other. Imagine 1 deleted and 1 added, or 1 deleted and 2 added.

The logic you want is:  does the peer cache have any peers that are not in the peer listing?

The quickest way to fix this would be to *always* call `handlePeerDelete()`, since it looks harmless to call even if nothing was deleted. We'll need to optimize the data structures and how the check is done later on. Right now, it looks like O(n^2). but that's not very important right now.